### ISSUE TITLE
Unknown Lexem type: ATTRIB_NAME

### ISSUE THREAD

**Issue by rohanabraham9** [nirgit/simple-xml-to-json#22]
Your code is so clean and a pleasure to read ! Wonderful work.
 
The following xml -

```xml
<p style="color: white;">This is a sentence <b>with</b> one word in bold</p>
```

throws the above error. On the other hand - 

```xml
<p style="color: white;">This is a sentence <b>with</b></p>
```
does not throw an error, but the content is wierd.

**Comment by rohanabraham9** [comment]
Perhaps the error is being thrown because this is an example of `semi structured xml` (see [here](https://www.xml.com/pub/a/2006/05/31/converting-between-xml-and-json.html)); and I am using the parser outside its intended use ?

**Comment by nirgit** [comment]
1. thanks for the kind words, i appreciate it!
2. you're right. the parser does not really support semistructured XML, and that is why it blows up. 
3. it'd be definitely be something to consider supporting.

I assume, an XML such as the one you tried to parse: 
`<p style="color: white;">This is a sentence <b>with</b> one word in bold</p>`

should get translated into a JSON that would look something like:
{
    "p": {
        "style": "color: white;",
        "children": [
                {
                    "$": {    content: "This is a sentence "    }
                },
                {
                    "b": {    content: "with"    }
                },
                {
                    "$": {    content: " one word in bold"    }
                }
         ]
    }
}

**Comment by rohanabraham9** [comment]
Yes ! Though, what are your thoughts on skipping the "$" altogether? e.g.

```json
{
    "p": {
        "style": "color: white;",
        "children": [
                {
                    "content": "This is a sentence "    
                },
                {
                    "b": {    "content": "with"    }
                },
                {
                    "content": " one word in bold"    
                }
         ]
    }
}
```
I feel it reads a little better. no ?

**Comment by nirgit** [comment]
fyi @rohanabraham9 
https://github.com/nirgit/simple-xml-to-json/pull/25

**Comment by rohanabraham9** [comment]
Thank you !On Apr 23, 2023, at 3:02 PM, Nir Moav ***@***.***> wrote:﻿
Closed #22 as completed.
